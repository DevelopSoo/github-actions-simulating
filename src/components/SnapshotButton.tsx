// src/components/SnapshotButton.tsx

export default function SnapshotButton({
  children,
  disabled,
  isLoading,
}: {
  children: React.ReactNode;
  disabled?: boolean;
  isLoading?: boolean;
}) {
  return (
    <button
      className={`bg-blue-500 ${
        disabled ? "cursor-not-allowed opacity-50" : ""
      } ${isLoading ? "animate-pulse" : ""}`}
    >
      {children}
    </button>
  );
}
