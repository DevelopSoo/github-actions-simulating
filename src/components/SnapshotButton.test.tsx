// src/components/SnapshotButton.test.tsx

import { render } from "@testing-library/react";
import SnapshotButton from "./SnapshotButton";

describe("Button 컴포넌트가 올바르게 렌더링됩니다.", () => {
  test("기본 상태", () => {
    const { container } = render(<SnapshotButton>클릭</SnapshotButton>);
    expect(container).toMatchSnapshot();
  });

  test("비활성화 상태", () => {
    const { container } = render(
      <SnapshotButton disabled>클릭</SnapshotButton>,
    );
    expect(container).toMatchSnapshot();
  });

  test("로딩 상태", () => {
    const { container } = render(
      <SnapshotButton isLoading>클릭</SnapshotButton>,
    );
    expect(container).toMatchSnapshot();
  });
});
