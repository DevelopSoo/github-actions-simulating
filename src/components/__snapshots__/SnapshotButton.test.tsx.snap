// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 컴포넌트가 올바르게 렌더링됩니다. 기본 상태 1`] = `
<div>
  <button
    class="bg-blue-500  "
  >
    클릭
  </button>
</div>
`;

exports[`Button 컴포넌트가 올바르게 렌더링됩니다. 로딩 상태 1`] = `
<div>
  <button
    class="bg-blue-500  animate-pulse"
  >
    클릭
  </button>
</div>
`;

exports[`Button 컴포넌트가 올바르게 렌더링됩니다. 비활성화 상태 1`] = `
<div>
  <button
    class="bg-blue-500 cursor-not-allowed opacity-50 "
  >
    클릭
  </button>
</div>
`;
